### PR TITLE
gdal: update to 3.5.1

### LINKED ISSUE #12614: [gdal] Missing MSODBCSQL17.DLL
MSODBCSQL

This library is provided by Microsoft at https://docs.microsoft.com/fr-fr/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16

Would it be acceptable to disable the use of ODBC with [GDAL_USE_ODBC = Off](https://github.com/OSGeo/gdal/search?q=GDAL_USE_ODBC)?

### PR DESCRIPTION
**Pull request by MehdiChinoune**
Fixes #12614

### CODE REVIEW

**Comment by lazka** [comment]
Needs rebuilds due to DLL name change:


```
mingw-w64-postgis
mingw-w64-pdal
mingw-w64-OpenSceneGraph
mingw-w64-osgearth
mingw-w64-vtk
```

File: mingw-w64-gdal/PKGBUILD [hunk 1]
```diff
@@ -110,6 +110,8 @@ build() {
       -DBUILD_SHARED_LIBS=ON \
       -DCMAKE_CXX_STANDARD=17 \
       -DBUILD_PYTHON_BINDINGS=OFF \
+      -DGDAL_USE_INTERNAL_LIBS=OFF \
+      -DGDAL_USE_MSSQL_ODBC=OFF \
```
Comment (filnet) [inline mingw-w64-gdal/PKGBUILD:114]: Why not disable all ODBC stuff with [GDAL_USE_ODBC](https://github.com/OSGeo/gdal/search?q=GDAL_USE_ODBC) ?

Comment (MehdiChinoune) [reply mingw-w64-gdal/PKGBUILD:114]: Why?

Comment (filnet) [reply mingw-w64-gdal/PKGBUILD:114]: Don't know :)

**Comment by MehdiChinoune** [comment]
CLANG64 job succeeded and the others timed out, Is OK to merge?

**Comment by lazka** [comment]
yes, let's try :+1:

**Pull request merged**